packages/grid/x-data-grid/src/hooks/features/columns/useGridColumnSpanning.ts

```diff
@@ -38,9 +38,19 @@ export const useGridColumnSpanning = (apiRef: React.MutableRefObject<GridApiComm
       rowId: GridRowId;
       minFirstColumnIndex: number;
       maxLastColumnIndex: number;
+      includeHiddenColumns?: boolean;
     }) => {
-      const { columnIndex, rowId, minFirstColumnIndex, maxLastColumnIndex } = params;
-      const visibleColumns = apiRef.current.getVisibleColumns();
+      const {
+        columnIndex,
+        rowId,
+        minFirstColumnIndex,
+        maxLastColumnIndex,
+        includeHiddenColumns = false,
+      } = params;
+
+      const visibleColumns = includeHiddenColumns
```

Comment:
Wouldn't be simpler to pass the columns to use as a param?

In https://github.com/mui/mui-x/blob/29a1f55ba05a77cd7276608c6ff8fc7516300f21/packages/grid/x-data-grid-premium/src/hooks/features/export/useGridExcelExport.tsx#L36 we already calculate which columns to use. We're doing it again inside `apiRef.current.unstable_calculateColSpan`. Both places where this method is called we have the columns nearby.

```diff
diff --git a/packages/grid/x-data-grid-premium/src/hooks/features/export/serializer/excelSerializer.ts b/packages/grid/x-data-grid-premium/src/hooks/features/export/serializer/excelSerializer.ts
index e58808630..8dc2a399f 100644
--- a/packages/grid/x-data-grid-premium/src/hooks/features/export/serializer/excelSerializer.ts
+++ b/packages/grid/x-data-grid-premium/src/hooks/features/export/serializer/excelSerializer.ts
@@ -65,6 +65,7 @@ const serializeRow = (
     rowId: id,
     minFirstColumn: 0,
     maxLastColumn: columns.length - 1,
+    columns,
   });
 
   columns.forEach((column, colIndex) => {
diff --git a/packages/grid/x-data-grid/src/hooks/features/virtualization/useGridVirtualScroller.tsx b/packages/grid/x-data-grid/src/hooks/features/virtualization/useGridVirtualScroller.tsx
index cf3ccb353..18a0e5f97 100644
--- a/packages/grid/x-data-grid/src/hooks/features/virtualization/useGridVirtualScroller.tsx
+++ b/packages/grid/x-data-grid/src/hooks/features/virtualization/useGridVirtualScroller.tsx
@@ -400,7 +400,12 @@ export const useGridVirtualScroller = (props: UseGridVirtualScrollerProps) => {
       const row = currentPage.rows[i];
       renderedRows.push(row);
 
-      apiRef.current.unstable_calculateColSpan({ rowId: row.id, minFirstColumn, maxLastColumn });
+      apiRef.current.unstable_calculateColSpan({
+        rowId: row.id,
+        minFirstColumn,
+        maxLastColumn,
+        columns: visibleColumns,
+      });
     }
 
     const [initialFirstColumnToRender, lastColumnToRender] = getRenderableIndexes({
```

Reply:
Thanks, it makes total sense!